app/connectors/UpscanConnector.scala

```diff
@@ -17,28 +17,41 @@
 package connectors
 
 import config.ApplicationConfig
+
 import javax.inject.{Inject, Named}
 import models.upscan.UpscanInitiateRequest
 import models.upscan.{PreparedUpload, UpscanInitiateResponse}
-import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.http.HttpClient
+import play.api.libs.json.Json
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, StringContextOps, UpstreamErrorResponse}
 import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.http.client.HttpClientV2
+
 import scala.concurrent.{ExecutionContext, Future}
 
 class UpscanConnector @Inject()(
                                  configuration: ApplicationConfig,
-                                 httpClient: HttpClient,
+                                 httpClient: HttpClientV2,
                                  @Named("appName") appName: String
                                )(implicit ec: ExecutionContext) {
 
   private val upscanInitiateHost: String = configuration.upscanInitiateHost
   private[connectors] val upscanInitiatePath: String = "/upscan/v2/initiate"
   private val upscanInitiateUrl: String = upscanInitiateHost + upscanInitiatePath
 
-  def getUpscanFormData(body: UpscanInitiateRequest)
-                       (implicit hc: HeaderCarrier): Future[UpscanInitiateResponse] = {
-    httpClient.POST[UpscanInitiateRequest, PreparedUpload](upscanInitiateUrl, body).map {
-      _.toUpscanInitiateResponse
+  def getUpscanFormData(body: UpscanInitiateRequest)(implicit hc: HeaderCarrier): Future[UpscanInitiateResponse] = {
+    httpClient.post(url"$upscanInitiateUrl")
+      .withBody(Json.toJson(body))
+      .execute[HttpResponse]
+      .flatMap { httpResponse =>
```

Comment:
This code looks a bit different to previous - where is it coming from?

Reply:
It was added to debug the reponses forgot to revert.